platform/email/src/main/java/org/motechproject/email/settings/SettingsDto.java

```diff
@@ -46,19 +60,43 @@ public Properties toProperties() {
         Properties properties = new Properties();
         properties.put(MAIL_HOST_PROPERTY, host);
         properties.put(MAIL_PORT_PROPERTY, port);
+        properties.put(MAIL_USERNAME_PROPERTY, username);
+        properties.put(MAIL_PASSWORD_PROPERTY, password);
         properties.put(MAIL_LOG_ADDRESS_PROPERTY, logAddress);
         properties.put(MAIL_LOG_SUBJECT_PROPERTY, logSubject);
         properties.put(MAIL_LOG_BODY_PROPERTY, logBody);
         properties.put(MAIL_LOG_PURGE_ENABLE_PROPERTY, logPurgeEnable);
         properties.put(MAIL_LOG_PURGE_TIME_PROPERY, logPurgeTime);
         properties.put(MAIL_LOG_PURGE_TIME_MULTIPLIER_PROPERTY, logPurgeTimeMultiplier);
-
         return properties;
     }
 
-    public SettingsDto(String host, String port, String logAddress, String logSubject, String logBody, String logPurgeEnable) {
+    public SettingsDto(String host, String port, String user, String password, Properties additionalProperties, // NO CHECKSTYLE More than 7 parameters (found 8).
+                       String logAddress, String logSubject, String logBody, String logPurgeEnable,
+                       String logPurgeTime, String logPurgeTimeMultiplier) {
+        this.host = host;
+        this.port = port;
+        this.username = user;
+        this.password = password;
+        Map<String, String> props = new HashMap<>();
+        for (Entry<Object, Object> entry : additionalProperties.entrySet()) {
```

Comment:
NPE when additionalProperties is null; Also can you refactor all of these constructors to invoke more complex one and have only one constructor set the actual values?